Realm/RLMSyncUser.mm

```diff
@@ -400,26 +400,24 @@ - (void)acceptOfferForToken:(NSString *)token
         callback(nil, make_permission_error_change(@"A permission offer cannot be accepted by an invalid user."));
         return;
     }
-    auto cb = [callback](util::Optional<std::string> raw_url, std::exception_ptr ptr) {
+    NSURLComponents *baseURL = [NSURLComponents componentsWithString:[self.authenticationServer absoluteString]];
+    if ([baseURL.scheme isEqualToString:@"http"]) {
+        baseURL.scheme = @"realm";
+    } else if ([baseURL.scheme isEqualToString:@"https"]) {
+        baseURL.scheme = @"realms";
+    } else {
+        @throw RLMException(@"A ROS server URL must begin with http:// or https://.");
```

Comment:
Shouldn't we be validating the protocol when creating the user rather than at this later point? Can we even reach this point if the auth server URL wasn't HTTP / HTTPS?

We should use "Realm Object Server" rather than ROS in error messages, and we should distinguish between authentication and Realm URLs (http(s) vs realm(s)).

Reply:
Yeah, you're right. If you are a normal user, you by definition have to log in successfully first to even have a user object. If you are an admin token user we can do the validation there.